pkg/v1/remote/write_test.go

```diff
@@ -606,3 +608,139 @@ func TestWriteWithErrors(t *testing.T) {
 		t.Errorf("Write(); (-want +got) = %s", diff)
 	}
 }
+
+func TestScopesForUploadingImage(t *testing.T) {
+	serverHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
It looks like you're spinning up a fake server just to get a valid URL. I'd remove these lines and just hard-code the references to be, e.g.:
`example.com/sample/sample:latest`
`example.com/sample/another_repo1:latest`
`example.com/sample/another_repo2:latest`

I think that will be easier to follow, since the string concatenation obfuscates what's going on a bit.